### ISSUE TITLE
CUDA

### ISSUE THREAD

**Issue by SeBin7** [SeBin7/onednn-xpu-fusedops-bench#10]
root@9555d6fc338e:/workspace# python - <<'PY'
import numpy as np, torch
print("numpy :", np.__version__)
try:
    import linear_gelu_cuda as lg
    print("linear_gelu_cuda ok:", [s for s in dir(lg) if "gelu" in s][:5])
except Exception as e:
    print("linear_gelu_cuda import failed:", e)
PY
numpy : 1.26.4
linear_gelu_cuda ok: ['bias_gelu_backward', 'bias_gelu_forward']
root@9555d6fc338e:/workspace# python - <<'PY'
import torch
from torch.profiler import profile, record_function, ProfilerActivity
from xpu_bench.algos.mlp import build_mlp_model

device=torch.device("cuda")
cfg=dict(in_dim=1536, hidden=1536, layers=8, out_dim=1536, device=device)
m_fused = build_mlp_model(**cfg, use_fused=True).to(device).eval()
x = torch.randn(8,1536,device=device)

with profile(activities=[ProfilerActivity.CPU, ProfilerActivity.CUDA], record_shapes=True) as prof:
    with record_function("warmup"):
        for _ in range(20):
            y = m_fused(x); torch.cuda.synchronize()
    with record_function("measure"):
        for _ in range(50):
            y = m_fused(x); torch.cuda.synchronize()

PYint(prof.key_averages().table(sort_by="cuda_time_total", row_limit=15))
/usr/local/lib/python3.10/dist-packages/torch/autograd/profiler.py:215: UserWarning: The attribute `use_cuda` will be deprecated soon, please use ``use_device = 'cuda'`` instead.
  warn(
STAGE:2025-10-13 21:45:07 1439:1439 ActivityProfilerController.cpp:316] Completed Stage: Warm Up
STAGE:2025-10-13 21:45:07 1439:1439 ActivityProfilerController.cpp:322] Completed Stage: Collection
STAGE:2025-10-13 21:45:07 1439:1439 ActivityProfilerController.cpp:326] Completed Stage: Post Processing
-------------------------------------------------------  ------------  ------------  ------------  ------------  ------------  ------------  ------------  ------------  ------------  ------------  
                                                   Name    Self CPU %      Self CPU   CPU total %     CPU total  CPU time avg     Self CUDA   Self CUDA %    CUDA total  CUDA time avg    # of Calls  
-------------------------------------------------------  ------------  ------------  ------------  ------------  ------------  ------------  ------------  ------------  ------------  ------------  
                                           aten::linear         2.09%       1.838ms        50.06%      44.139ms      70.062us       0.000us         0.00%      54.553ms      86.592us           630  
                                           aten::matmul         1.32%       1.166ms        35.42%      31.231ms      55.769us       0.000us         0.00%      48.107ms      85.906us           560  
                                               aten::mm        13.78%      12.149ms        34.10%      30.065ms      53.687us      48.107ms        38.14%      48.107ms      85.906us           560  
                                  sgemm_32x32x32_NT_vec         0.00%       0.000us         0.00%       0.000us       0.000us      48.107ms        38.14%      48.107ms      85.906us           560  
                                                measure         0.00%       0.000us         0.00%       0.000us       0.000us      46.636ms        36.97%      46.636ms      46.636ms             1  
                                                measure        16.44%      14.497ms        53.05%      46.773ms      46.773ms       0.000us         0.00%      39.719ms      39.719ms             1  
                                                 warmup         0.00%       0.000us         0.00%       0.000us       0.000us      23.724ms        18.81%      23.724ms      23.724ms             1  
                                                 warmup         7.24%       6.382ms        46.95%      41.393ms      41.393ms       0.000us         0.00%      16.226ms      16.226ms             1  
                                            aten::addmm         3.34%       2.946ms         6.95%       6.129ms      87.552us       6.445ms         5.11%       6.445ms      92.078us            70  
void gemmSN_TN_kernel<float, 128, 16, 2, 4, 8, 9, fa...         0.00%       0.000us         0.00%       0.000us       0.000us       6.445ms         5.11%       6.445ms      92.078us            70  
                                            _BiasGELUFn         8.85%       7.804ms        19.38%      17.090ms      30.517us       1.217ms         0.97%       1.217ms       2.174us           560  
void (anonymous namespace)::bias_gelu_fwd_kernel<flo...         0.00%       0.000us         0.00%       0.000us       0.000us       1.217ms         0.97%       1.217ms       2.174us           560  
                                  cudaDeviceSynchronize         6.87%       6.060ms         6.87%       6.060ms      85.359us     175.457us         0.14%     175.457us       2.471us            71  
                                                aten::t         2.84%       2.504ms         5.60%       4.941ms       7.843us       0.000us         0.00%       0.000us       0.000us           630  
                                        aten::transpose         1.79%       1.577ms         2.76%       2.437ms       3.869us       0.000us         0.00%       0.000us       0.000us           630  
-------------------------------------------------------  ------------  ------------  ------------  ------------  ------------  ------------  ------------  ------------  ------------  ------------  
Self CPU time total: 88.168ms
Self CUDA time total: 126.130ms
